src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -11216,6 +11221,23 @@ void local()
             Assert.False(verifier.HasLocalsInit("C.<M>g__local|0_0"));
         }
 
+        [Fact, WorkItem(49434, "https://github.com/dotnet/roslyn/issues/49434")]
+        public void SkipLocalsInit_Module_TopLevelStatements()
+        {
+            var source = @"
+using System.Runtime.CompilerServices;
+using System;
+
+[module: SkipLocalsInit]
+
+var w = 1;
+w = w + w + w;
+";
+
+            var verifier = CompileAndVerifyWithSkipLocalsInit(source, TestOptions.UnsafeReleaseExe);
+            Assert.False(verifier.HasLocalsInit("<top-level-statements-entry-point>"));
```

Comment:
>"<top-level-statements-entry-point>" [](start = 48, length = 36)

Is this assert failing without the fix? This isn't really a qualified name.

Reply:
Yup, I actually ran the test without `SkipLocalsInit` as a sanity test. (I didn't add a separate test for this since it doesn't seem like inverse tests are common in Roslyn, can add one if you want.)

`WellKnownAttributeTestsUtil.HasLocalsInit` uses `CompilationTestDataExtensions.GetMethodData` to find the method, which [throws if it isn't found](https://github.com/dotnet/roslyn/blob/f8a72e53fdd622735ef022f69c34d25f276c26c2/src/Compilers/Test/Core/Compilation/CompilationTestDataExtensions.cs#L74). The reason it isn't `<Program>$.<Main>$` is that the method lookup is keyed using [`ToDisplayString`](https://github.com/dotnet/roslyn/blob/f8a72e53fdd622735ef022f69c34d25f276c26c2/src/Compilers/Core/Portable/CodeGen/CompilationTestData.cs#L111) which is [hard-coded to return `<top-level-statements-entry-point>` for the synthesized `Main`](https://github.com/dotnet/roslyn/blob/f8a72e53fdd622735ef022f69c34d25f276c26c2/src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs#L231-L236).

Reply:
> I didn't add a separate test for this since it doesn't seem like inverse tests are common in Roslyn, can add one if you want.

Yes, please add a test asserting that local init is present without the attribute.

---
In reply to: [525524563](https://github.com/dotnet/roslyn/pull/49435#discussion_r525524563) [](ancestors = 525524563)

Reply:
Done!